### PR TITLE
Return type hints

### PR DESCRIPTION
**Pull request by rarila**
- Fix missing native return typehints
- Fix missing any return typehints
- Remove useless return typehints in annotations
- Consistent formatting of return typehints
- Add sniffs checking return typehints

### CODE REVIEW

File: src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/ArithmeticOpAnalyzer.php [hunk 1]
```diff
@@ -733,7 +733,7 @@ private static function analyzeOperands(
                         }
                     } else {
                         $result_type = Type::combineUnionTypes(
-                            $always_positive ? Type::getPositiveInt(true) : Type::getInt(true),
+                            $always_positive ? Type::getPositiveInt(true): Type::getInt(true),
```
Comment (weirdan) [inline src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/ArithmeticOpAnalyzer.php:736]: Was it forced by slevomat CS?

File: tests/ProjectCheckerTest.php [hunk 2]
```diff
@@ -124,8 +124,7 @@ public function testAfterCodebasePopulatedIsInvoked(): void
             /** @var bool */
             public static $called = false;
 
-            /** @return void */
-            public static function afterCodebasePopulated(AfterCodebasePopulatedEvent $event)
+            public static function afterCodebasePopulated(AfterCodebasePopulatedEvent $event): void
```
Comment (weirdan) [inline tests/ProjectCheckerTest.php:127]: I suspect it'll fail on PHP <7.4

File: src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/ArithmeticOpAnalyzer.php (hunk 1 above)
Comment (rarila) [reply src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/ArithmeticOpAnalyzer.php:736]: No, I’m afraid, it was me. Don’t drink and code :smirk:

File: tests/ProjectCheckerTest.php (hunk 2 above)
Comment (rarila) [reply tests/ProjectCheckerTest.php:127]: Good catch! I thought I had all of these. I guess this one slipped through.
Arghl, found another one…

Comment (rarila) [reply tests/ProjectCheckerTest.php]: Always good to have a good an excellent reviewer :relieved:
Those interfaces shouldn’t differ in using or not using a return type.

File: src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/ArithmeticOpAnalyzer.php (hunk 1 above)
Comment (weirdan) [reply src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/ArithmeticOpAnalyzer.php]: There were actually several of those.

Comment (rarila) [reply src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/ArithmeticOpAnalyzer.php]: And yet again you are right

**Review by weirdan — APPROVED** [review]

**Comment by weirdan** [comment]
Thanks!

**Pull request merged**